tests/optional_scalars2_generated.rs

```diff
@@ -18,69 +18,74 @@ pub mod optional_scalars {
   extern crate flatbuffers;
   use self::flatbuffers::EndianScalar;
 
-#[allow(non_camel_case_types)]
-#[repr(i8)]
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
-pub enum OptionalByte {
-  None = 0,
-  One = 1,
-  Two = 2,
-
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct OptionalByte(pub i8);
+#[allow(non_upper_case_globals)]
+impl OptionalByte {
+  pub const ENUM_MIN: i8 = 0;
+  pub const ENUM_MAX: i8 = 2;
+  pub const None: Self = Self(0);
+  pub const One: Self = Self(1);
+  pub const Two: Self = Self(2);
+  pub const ENUM_VALUES: &'static [Self] = &[
+    Self::None,
+    Self::One,
+    Self::Two,
+  ];
+  /// Returns the variant's name or "" if unknown.
+  pub fn variant_name(self) -> &'static str {
+    match self {
+      Self::None => "None",
+      Self::One => "One",
+      Self::Two => "Two",
+      _ => "",
```

Comment:
This is a bad design, we should choose between unreachable!() and returning an Option. @rw opinions?

Reply:
Done, we're using Option.